### DISCUSSION TITLE
Removal of the deprecated /controlnet/txt2img and /controlnet/img2img routes

### DISCUSSION THREAD

**Discussion by ljleb** [Mikubill/sd-webui-controlnet#1132] [General]
We are going to remove these custom inference routes that were deprecated a while ago:

- /controlnet/txt2img
- /controlnet/img2img

I suggest to remove them in about a week, so on May 9. I'm just suggesting this timeline in case code from other repositories relying on them remains to be updated. We can remove them earlier or later if other collaborators have a different opinion on the subject, I don't mind.

The reason for the removal is that maintaining the route parameters takes time that could be used for more productive endeavours.

If anyone has objections to the removal, please let us know so that we can discuss and plan together for important use-cases.

**Comment by lllyasviel** [comment]
please do it rightnow.
after that perhaps I will write a bit more tests for some other video processing projects

**Reply by ljleb** [reply]
alright

**Reply by ljleb** [reply]
The deprecated routes are removed now. I suggest we revert the changes if anyone has important objections. We should wait for users when we are introducing breaking changes, as otherwise it can create incompatibilities between extensions or depending code.

**Reply by ghost** [reply]
If you use the same endpoint for txt2img as without controlnet, how is the API supposed to know to use controlnet? I'm trying to use the Web API, and i configured my json input correctly per the wiki, but it's only using the prompt and generating an image, doesn't use controlnet whatsoever. I have it set to use canny model, and i can tell in the console it's not even loading the canny model. Can you please advise? Thanks

**Reply by ljleb** [reply]
If you are not on latest on both the webui and controlnet, please update both. The deprecated routes were setup to fall back on the alwayson script args property in the json payload, so there really should be no difference in behavior between the deprecated and webui inference routes.

**Reply by ljleb** [reply]
There was an issue related to this but it seems it was closed. Please open an issue if the webui route does not work with controlnet, including the appropriate information so we can resolve the issue.

**Reply by ghost** [reply]
Appreciate the reply. I have updated controlnet and webui, but still no dice. I created an issue and included my json formatting. Appreciate any help you can offer. Thanks!

**Reply by Shonrain** [reply]
> If you use the same endpoint for txt2img as without controlnet, how is the API supposed to know to use controlnet? I'm trying to use the Web API, and i configured my json input correctly per the wiki, but it's only using the prompt and generating an image, doesn't use controlnet whatsoever. I have it set to use canny model, and i can tell in the console it's not even loading the canny model. Can you please advise? Thanks

you can use sdapi/v1/txt2img,here is the json：

```javascript
{
  ...,
	"alwayson_scripts": {
     "controlnet": {
       "args":[
         {
           	"input_image": "",
           "mask": "",
          "module": "scribble_xdog",
          "model": "control_v11p_sd15_scribble [d4ba51ff]",
          "weight": 1,
          "resize_mode": "Crop and Resize",
          "lowvram": false,
          "processor_res": 64,
          "threshold_a": 64,
          "threshold_b": 64,
          "guidance": 1,
          "guidance_start": 0,
          "guidance_end": 1,
          "guessmode": true,
          "pixel_perfect": false
         } 
       ]
     }
  }
}
```

**Reply by amota511** [reply]
how might we utilize multiple controlnet units using the above json? I'm adding just another object to args but it is not utilizing the second controlnet unit

**Reply by CryptoCoinTrader** [reply]
I was able able to install controlnet however :(     installed stable diffusion Auto1111, and about 12 extensions, but .....I'm starting to believe the extension for controlnet is removed from the Github repository maybe recently, am I going nuts? Are you seeing it the extension on Github? I'm a newb and know just enough to get in trouble, could use help please
